### PR TITLE
Fix bug in Gemma2 HuggingFace weight mapping

### LINKED ISSUE #2850: Gemma2 HuggingFace weight mapping wrong, loading/generation broken
Gemma2 is currently broken when using HuggingFace checkpoints, as there is a bug in the MLP norm weight mapping between HF and torchtune models:

Currently you have (https://github.com/pytorch/torchtune/blob/main/torchtune/models/gemma2/_convert_weights.py#L31):

```
"model.layers.{}.post_feedforward_layernorm.weight": "layers.{}.mlp_norm.scale",
"model.layers.{}.pre_feedforward_layernorm.weight": "layers.{}.mlp_scale.scale",
```

What you should have:

```
"model.layers.{}.post_feedforward_layernorm.weight": "layers.{}.mlp_scale.scale",
"model.layers.{}.pre_feedforward_layernorm.weight": "layers.{}.mlp_norm.scale",
```

---------------------

To verify, simple generation config:

```
# TODO: set this
checkpoint_dir: /tmp/gemma2-2b-it
output_dir: /tmp

tokenizer:
  _component_: torchtune.models.gemma.gemma_tokenizer
  path: ${checkpoint_dir}/tokenizer.model

seed: 123

model:
  _component_: torchtune.models.gemma2.gemma2_2b

checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: ${checkpoint_dir}
  checkpoint_files: [
    model-00001-of-00002.safetensors,
    model-00002-of-00002.safetensors,
  ]
  recipe_checkpoint: null
  output_dir: ${output_dir}/gemma2-2b-it
  model_type: GEMMA2
resume_from_checkpoint: False

device: cuda
dtype: bf16

quantizer: null
enable_kv_cache: False

prompt:
  system: null
  user: "What is pytorch?"
max_new_tokens: 128
temperature: 1.0
top_k: 64
```

Running currently:

```
tune run generate --config gemma2-2b.yaml

INFO:torchtune.utils._logging:Running InferenceRecipe with resolved config:

...

DEBUG:torchtune.utils._logging:Setting manual seed to local seed 123. Local seed is seed + rank = 123 + 0
INFO:torchtune.utils._logging:Model is initialized with precision torch.bfloat16.
INFO:torchtune.utils._logging:What is pytorch?tagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagexttagext
INFO:torchtune.utils._logging:Time for inference: 5.19 sec total, 24.66 tokens/sec
INFO:torchtune.utils._logging:Bandwidth achieved: 129.16 GB/s
INFO:torchtune.utils._logging:Memory used: 5.72 GB
```

With fix:

```
tune run generate --config gemma2-2b.yaml

INFO:torchtune.utils._logging:Running InferenceRecipe with resolved config:

...

DEBUG:torchtune.utils._logging:Setting manual seed to local seed 123. Local seed is seed + rank = 123 + 0
INFO:torchtune.utils._logging:Model is initialized with precision torch.bfloat16.
INFO:torchtune.utils._logging:What is pytorch?
Let's break down PyTorch and its significance in the world of machine learning.

**What is PyTorch?**

PyTorch is a powerful open-source machine learning library developed by Facebook's AI Research team. This library has gained immense popularity and is widely used in academia and industry for its Pythonic capabilities, flexibility, and ease of use.

**Key Features:**

* **Dynamic Computational Graph:** Unlike TensorFlow, PyTorch uses a dynamic computational graph. This means that the graph is built as you run the program, allowing for more flexibility during development.
* **Automatic Differentiation:** PyTorch can automatically
INFO:torchtune.utils._logging:Time for inference: 5.24 sec total, 24.42 tokens/sec
INFO:torchtune.utils._logging:Bandwidth achieved: 127.88 GB/s
INFO:torchtune.utils._logging:Memory used: 5.72 GB
```

### PR DESCRIPTION
**Pull request by johnglover**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

Fixes #2850

#### Changelog
What are the changes made in this PR?
* Fix bug in Gemma2 HF weight mapping (pre/post MLP weights are the wrong way round).

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [x] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

See the linked issue for full details (description and config for reproduction)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [x] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Review by felipemello1 — APPROVED** [review]
lgtm1 ty!
